test/tests/functional/pbs_reliable_job_startup.py

```diff
@@ -5435,3 +5454,214 @@ def test_t20(self):
+                      self.nC, self.nD, self.nE, self.nEv0]
+        for vn in vnode_list:
+            self.momA.log_match("Job;%s;launch: found vnode_list[%s]" % (
+                                jid, vn), n=10)
+
+        # Check vnode_list_fail[] parameter in execjob_launch hook
+        vnode_list_fail = [self.nC, self.nE, self.nEv0]
+        for vn in vnode_list_fail:
+            self.momA.log_match("Job;%s;launch: found vnode_list_fail[%s]" % (
+                                jid, vn), n=10)
+
+        # Check result of pbs.event().job.release_nodes(keep_select) call
+        self.momA.log_match("Job;%s;launch: job.exec_vnode=%s" % (
+            jid, self.job1v4_exec_vnode), n=10)
+
+        self.momA.log_match("Job;%s;launch: job.schedselect=%s" % (
+            jid, self.job1v4_schedselect), n=10)
+
+        self.momA.log_match("Job;%s;pruned from exec_vnode=%s" % (
+            jid, self.job1_iexec_vnode), n=10)
+
+        self.momA.log_match("Job;%s;pruned to exec_vnode=%s" % (
+            jid, self.job1v4_exec_vnode), n=10)
+
+        # Check accounting_logs
+        self.match_accounting_log('S', jid, self.job1_iexec_host_esc,
+                                  self.job1_iexec_vnode_esc, "10gb", 13, 5,
+                                  self.job1_place,
+                                  self.job1_isel_esc)
+
+        self.match_accounting_log('s', jid, self.job1v4_exec_host_esc,
+                                  self.job1v4_exec_vnode_esc,
+                                  "6gb", 8, 3,
+                                  self.job1_place,
+                                  self.job1v4_sel_esc)
+
+        self.momA.log_match("Job;%s;task.+started, hostname" % (jid,),
+                            n=10, max_attempts=60, interval=2, regexp=True)
+
+        self.momA.log_match("Job;%s;copy file request received" % (jid,),
+                            n=10, max_attempts=10, interval=2)
+
+        # validate output
+        expected_out = """/var/spool/pbs/aux/%s
+%s
+%s
+%s
+FIB TESTS
+pbsdsh -n 1 fib 37
+%d
+pbsdsh -n 2 fib 37
+%d
+fib 37
+%d
+HOSTNAME TESTS
+pbsdsh -n 0 hostname
+%s
+pbsdsh -n 1 hostname
+%s
+pbsdsh -n 2 hostname
+%s
+PBS_NODEFILE tests
+HOST=%s
+pbs_tmrsh %s hostname
+%s
+HOST=%s
+pbs_tmrsh %s hostname
+%s
+HOST=%s
+pbs_tmrsh %s hostname
+%s
+""" % (jid, self.momA.hostname, self.momB.hostname, self.momD.hostname,
+            self.fib37_value, self.fib37_value, self.fib37_value,
+            self.momA.hostname, self.momB.hostname, self.momD.hostname,
+            self.momA.hostname, self.momA.hostname, self.momA.hostname,
+            self.momB.hostname, self.momB.hostname, self.momB.hostname,
+            self.momD.hostname, self.momD.hostname, self.momD.hostname)
+
+        self.logger.info("expected out=%s" % (expected_out,))
+        job_out = ""
+        with open(job_output_file, 'r') as fd:
+            job_out = fd.read()
+            self.logger.info("job_out=%s" % (job_out,))
```

Comment:
Don't you need to wait until the job ends first?

Reply:
The job output file becomes available upon matching the following in line 5622:

self.momA.log_match("Job;%s;copy file request received" % (jid,),
                            n=10, max_attempts=10, interval=2)